python/ray/serve/_private/http_proxy.py

```diff
@@ -535,19 +449,191 @@ async def __call__(self, scope, receive, send):
+                # Assuming 10 retries, 1.5x scaling means the last retry is 38x the
+                # initial backoff time, while 2x scaling means 512x the initial.
+                backoff_time_s *= 1.5
+                retries += 1
+                backoff = False
+        else:
+            error_message = f"Task failed with {HTTP_REQUEST_MAX_RETRIES} retries."
+            await Response(error_message, status_code=500).send(scope, receive, send)
+            return "500"
+
+        if isinstance(result, (starlette.responses.Response, RawASGIResponse)):
+            await result(scope, receive, send)
+            return str(result.status_code)
+        else:
+            await Response(result).send(scope, receive, send)
+            return "200"
+
+    async def proxy_asgi_receive(self, receive: Receive, queue: asyncio.Queue):
+        """TODO"""
+        disconnected = False
+        while not disconnected:
+            msg = await receive()
+            await queue.put(msg)
+            if msg["type"] in {"http.disconnect", "websocket.disconnect"}:
+                disconnected = True
+
+    async def send_request_to_replica_streaming(
+        self,
+        request_id: str,
+        handle: RayServeHandle,
+        scope: Scope,
+        receive: Receive,
+        send: Send,
+    ) -> str:
+        # Proxy the receive interface by placing the received messages on a queue.
+        # The downstream replica must call back into `receive_asgi_message` on this
+        # actor to receive the messages.
+        receive_queue = asyncio.Queue()
+        self.asgi_receive_queues[request_id] = receive_queue
+        proxy_asgi_receive_task = get_or_create_event_loop().create_task(
+            self.proxy_asgi_receive(receive, receive_queue)
+        )
+
+        status_code = ""
+        first_message_sent = False
+        try:
+            object_ref_generator = await handle.remote(
+                pickle.dumps(scope), self.self_actor_handle
+            )
+            async for obj_ref in object_ref_generator:
+                asgi_messages: List[Dict[str, Any]] = pickle.loads(await obj_ref)
+                for asgi_message in asgi_messages:
+                    if not first_message_sent:
+                        # HTTP responses begin with exactly one "http.response.start"
+                        # message containing the "status" field. Other response types
+                        # (e.g., WebSockets) may not.
+                        status_code = str(asgi_message.get("status", "UNKNOWN"))
+
+                    first_message_sent = True
+                    await send(asgi_message)
+        except Exception as e:
+            error_message = f"Unexpected error, traceback: {e}."
+            logger.warning(error_message)
+
+            # If the first message has been sent we cannot send a 500 status as it
+            # would violate ASGI protocol.
+            # TODO(edoakes): this may not be valid at all for WebSocket requests.
```

Comment:
Should we add a test for this?

Reply:
removed this logic entirely. we'll still return an `Internal Server Error` in this case